### PR TITLE
[Feature] Add cluster status check interface and scheduled tasks

### PR DESCRIPTION
**Pull request by AnemoneIndicum**
### Purpose
close: https://github.com/apache/paimon-webui/issues/317

### Tests

- `ClusterControllerTest#testGetHeartbeatActionFactory`
- `ClusterControllerTest#testCheckClusterHeartbeatStatus`
- `ClusterControllerTest#testCheckClusterHeartbeatOfFlinkSession` 
- `ClusterControllerTest#testCheckClusterHeartbeatOfFlinkSqlGateway`

### API and Format
```java
  @SaCheckPermission("system:cluster:check")
    @PostMapping("/check")
    public R<Void> check(@Validated @RequestBody ClusterInfo clusterInfo) {
        return clusterService.checkClusterHeartbeatStatus(clusterInfo) ? R.succeed() : R.failed();
    }
```

```java
  
/** Using to execute cluster heartbeat action. */
public interface HeartbeatAction {

    /**
     * Execute cluster action to obtain cluster status.
     *
     * @return cluster heartbeat entity.
     */
    HeartbeatEntity checkClusterHeartbeat();

    /**
     * Build a heartbeat entity representing an error based on the cluster status. This method is
     * used to generate a heartbeat object when the cluster status is abnormal, recording the
     * current time and the error status of the cluster.
     *
     * @param status The current status of the cluster, used to set the status field of the
     *     heartbeat entity.
     * @return Returns a completed heartbeat entity, including the current timestamp and status
     *     information.
     */
    default HeartbeatEntity buildResulHeartbeatEntity(HeartbeatStatus status) {
        return HeartbeatEntity.builder().status(status.name()).build();
    }
}
```

### CODE REVIEW

File: paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * Using to execute cluster action
+ */
+public interface ClusterAction {
+
+    /**
+     * Execute cluster action to obtain cluster status.
+     *
+     * @return cluster heartbeat entity
+     */
+    ImmutablePair<ClusterStatus, Long> checkClusterHeartbeat() throws Exception;
```
Comment (SteNicholas) [inline paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java:33]: Why use `ImmutablePair`?

File: paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    default ImmutablePair<ClusterStatus, Long> buildClusterHeartbeatOfSuccess() {
+       return ImmutablePair.of(ClusterStatus.RUNNING,System.currentTimeMillis());
+    }
+    default ImmutablePair<ClusterStatus, Long> buildClusterHeartbeatOfError() {
+        return ImmutablePair.of(ClusterStatus.UNKNOWN,System.currentTimeMillis());
+    }
+    enum ClusterStatus {
+        RUNNING,
```
Comment (SteNicholas) [inline paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java:42]: Is there any other status?

File: paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ */
+
+package org.apache.paimon.web.engine.flink.sql.gateway.client;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+
+/**
+ * Using to execute cluster action
+ */
+public interface ClusterAction {
```
Comment (SteNicholas) [inline paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java:26]: Please add this interface into description of this pull request.

File: paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    /**
+     * Execute cluster action to obtain cluster status.
+     *
+     * @return cluster heartbeat entity
+     */
+    ImmutablePair<ClusterStatus, Long> checkClusterHeartbeat() throws Exception;
+
+    default ImmutablePair<ClusterStatus, Long> buildClusterHeartbeatOfSuccess() {
```
Comment (SteNicholas) [inline paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java:35]: Does this include any heartbeat action?

File: paimon-web-gateway/src/main/java/org/apache/paimon/web/gateway/provider/ExecutorFactoryProvider.java [hunk 5]
```diff
@@ -34,7 +34,7 @@ public ExecutorFactoryProvider(ExecutionConfig executionConfig) {
 
     public ExecutorFactory getExecutorFactory(EngineType type) {
         switch (type) {
-            case FLINK:
+            case FLINK_SQL_GATEWAY:
```
Comment (SteNicholas) [inline paimon-web-gateway/src/main/java/org/apache/paimon/web/gateway/provider/ExecutorFactoryProvider.java:37]: Is this change related with cluster status check?

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java (hunk 1 above)
Comment (AnemoneIndicum) [reply paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java:33]: Don't create other class.

File: paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java (hunk 4 above)
Comment (AnemoneIndicum) [reply paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java:35]: Yes,currently, only the cluster's liveness status is checked, and the specific implementation is carried out by different engines.

File: paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java (hunk 3 above)
Comment (AnemoneIndicum) [reply paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java:26]: I will do it later.

File: paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java (hunk 2 above)
Comment (AnemoneIndicum) [reply paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java]: I have added other status codes to ensure more scenarios are covered.

File: paimon-web-gateway/src/main/java/org/apache/paimon/web/gateway/provider/ExecutorFactoryProvider.java (hunk 5 above)
Comment (AnemoneIndicum) [reply paimon-web-gateway/src/main/java/org/apache/paimon/web/gateway/provider/ExecutorFactoryProvider.java:37]: To more clearly identify different Flink engines, in the cluster management page, Flink currently has session and sql gateway cluster types. This issue has been unified in the WeChat group a few days ago.

File: paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-common/src/main/java/org/apache/paimon/web/engine/flink/common/status/ClusterStatus.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.engine.flink.common.status;
+
+/** This enum represents the status of a cluster. */
+public enum ClusterStatus {
+    RUNNING,
+    UNKNOWN,
+    NETWORK_ERROR,
```
Comment (SteNicholas) [inline paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-common/src/main/java/org/apache/paimon/web/engine/flink/common/status/ClusterStatus.java:25]: I don't think `NETWORK_ERROR` and `EXECUTION_ERROR` are cluster status.

File: paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.engine.flink.sql.gateway.client;
+
+import org.apache.paimon.web.engine.flink.common.status.ClusterStatus;
+import org.apache.paimon.web.engine.flink.sql.gateway.model.HeartbeatEntity;
+
+/** Using to execute cluster action. */
+public interface ClusterAction {
```
Comment (SteNicholas) [inline paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java:25]: Could this use `HeartbeatAction`?

File: paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+     * Build a heartbeat entity representing an error based on the cluster status. This method is
+     * used to generate a heartbeat object when the cluster status is abnormal, recording the
+     * current time and the error status of the cluster.
+     *
+     * @param status The current status of the cluster, used to set the status field of the
+     *     heartbeat entity.
+     * @return Returns a completed heartbeat entity, including the current timestamp and status
+     *     information.
+     */
+    default HeartbeatEntity buildClusterHeartbeatOfError(ClusterStatus status) {
```
Comment (SteNicholas) [inline paimon-web-engine/paimon-web-engine-flink/paimon-web-engine-flink-sql-gateway/src/main/java/org/apache/paimon/web/engine/flink/sql/gateway/client/ClusterAction.java:44]: `OfError`? Which error?

File: paimon-web-gateway/src/main/java/org/apache/paimon/web/gateway/enums/EngineType.java [hunk 9]
```diff
@@ -21,7 +21,8 @@
 /** The {@code EngineType} enum defines the types of engines that can be supported. */
 public enum EngineType {
     SPARK,
-    FLINK;
+    FLINK_SQL_GATEWAY,
```
Comment (SteNicholas) [inline paimon-web-gateway/src/main/java/org/apache/paimon/web/gateway/enums/EngineType.java:24]: Please create another pull request to change this.

**Review by SteNicholas — CHANGES_REQUESTED** [review]

**Review by SteNicholas — APPROVED** [review]
